sympy/matrices/eigen.py

```diff
@@ -481,22 +510,36 @@ def _bidiagonalize(M, upper=True):
     upper : bool. Whether to do upper bidiagnalization or lower.
                 True for upper and False for lower.
 
+    method : string. ``lhc`` , ``householder``
+            ``householder`` , this method uses householder transformations
+            to bidiagonalize the input.
+            ``lhc`` , this method uses R-Bidiagonalization/LHC-Bidiagonalization
+            to bidiagonalize the input.
+
     References
     ==========
 
     1. Algorith 5.4.2, Matrix computations by Golub and Van Loan, 4th edition
     2. Complex Matrix Bidiagonalization : https://github.com/vslobody/Householder-Bidiagonalization
 
     """
+    method = method.lower()
+    if method not in ("lhc", "householder"):
+        raise ValueError("Invalid Method for Bidiagonalization")
 
     if type(upper) is not bool:
         raise ValueError("upper must be a boolean")
 
     if not upper:
-        return _eval_bidiag_hholder(M.H).H
-
-    return _eval_bidiag_hholder(M)
-
+        return M.H.bidiagonalize(method=method).H
+    if method == "lhc":
```

Comment:
Unfortunately, I don't find many things are optimized to `LHC` method you have implemented. 
And if it is basically computing the decomposition after another decomposition, it wouldn't be needed to made into a user API, because it's the matter of the execution order.
I'd expect some optimizations like skipping some eliminations.

Reply:
Currently, LHC is approximately 13x Faster than householder for a 3x3 Matrix, and I beleive many many times faster than Householder for 4x4 or above matrices.
```python3
In [10]: A = Matrix(3, 3, range(1, 10))

In [11]: %timeit A.bidiagonalize()
187 ms ± 13.5 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [12]: %timeit A.bidiagonalize(method="householder")
2.49 s ± 133 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```
I could figure out only one optimization and that is skipping the very first elimination and  It won't give a big reduction in time cost.
I will add a fix for that in the next commit.
If you beleive there are others, please do tell me.